src/content_scripts/autocorrect.js

```diff
@@ -0,0 +1,400 @@
+"use strict";
+
+const fractions = Object.freeze({
+    "¼": 1.0 / 4.0,
+    "½": 1.0 / 2.0,
+    "¾": 3.0 / 4.0,
+    "⅐": 1.0 / 7.0,
+    "⅑": 1.0 / 9.0,
+    "⅒": 1.0 / 10.0,
+    "⅓": 1.0 / 3.0,
+    "⅔": 2.0 / 3.0,
+    "⅕": 1.0 / 5.0,
+    "⅖": 2.0 / 5.0,
+    "⅗": 3.0 / 5.0,
+    "⅘": 4.0 / 5.0,
+    "⅙": 1.0 / 6.0,
+    "⅚": 5.0 / 6.0,
+    "⅛": 1.0 / 8.0,
+    "⅜": 3.0 / 8.0,
+    "⅝": 5.0 / 8.0,
+    "⅞": 7.0 / 8.0
+});
+
+const constants = Object.freeze({
+    "π": Math.PI,
+    "e": Math.E
+});
+
+// communication type
+// directly include magic constant as a workaround as we cannot import modules in content scripts due to https://bugzilla.mozilla.org/show_bug.cgi?id=1451545
+const AUTOCORRECT_CONTENT = "autocorrectContent";
+
+let insertedText; // Last insert text
+let deletedText; // Last deleted text
+let lastTarget; // Last target
+let lastCaretPosition; // Last caret position
+
+let quotes = true;
+let fracts = true;
+
+let autocorrections = {};
+
+let longest = 0;
+
+// Regular expressions
+let symbolpatterns = null;
+// Do not autocorrect for these patterns
+let apatterns = null;
+
+// Thunderbird
+const THUNDERBIRD = typeof messenger !== "undefined";
+
+// Chrome
+const CHROME = Object.getPrototypeOf(browser) !== Object.prototype;
+
+/**
+ * Get caret position.
+ *
+ * @param {Object} target
+ * @returns {number}
+ */
+function getCaretPosition(target) {
+    if (target.isContentEditable || THUNDERBIRD) {
+        target.focus();
+        const _range = document.getSelection().getRangeAt(0);
+        const range = _range.cloneRange();
+        const temp = document.createTextNode("\0");
+        range.insertNode(temp);
+        const caretposition = target.innerText.indexOf("\0");
+        temp.parentNode.removeChild(temp);
+        return caretposition;
+    } else {
+        return target.selectionStart;
+    }
+}
+
+/**
+ * Insert at caret.
+ * Adapted from: https://www.everythingfrontend.com/posts/insert-text-into-textarea-at-cursor-position.html
+ *
+ * @param {Object} target
+ * @param {string} atext
+ * @returns {void}
+ */
+function insertCaret(target, atext) {
```

Comment:
```suggestion
function insertAtCaret(target, atext) {
```

Otherwise this would mean we insert the caret itself. :blush: